demisto_sdk/dev_tools/spell_checker.py

```diff
@@ -0,0 +1,126 @@
+import yaml
+from typing import Set
+from spellchecker import SpellChecker
+
+from argparse import ArgumentDefaultsHelpFormatter
+from demisto_sdk.common.constants import KNOWN_WORDS
+from demisto_sdk.common.tools import print_error, print_color, LOG_COLORS
+
+
+# These are keys in a Demisto yml file which indicate that their values are visible to the user and
+# thus their spelling should be checked.
+USER_VISIBLE_LINE_KEYS = [
+    "description",
+    "display",
+    "name",
+    "comment"
+]
+
+
+class SpellCheck:
+    """Perform a spell check on the given .yml or .md file.
+        Attributes:
+            path (str): The path to the current file being checked.
+            known_words (str): The path to a file containing known words.
+            spellchecker (SpellChecker): The spell-checking object.
+            unknown_words (set): A set of unknown words found in the given file.
+    """
+
+    def __init__(self, checked_file_path: str, known_words: str):
+        self.checked_file_path = checked_file_path
+        self.spellchecker = SpellChecker()
+        self.unknown_words = set([])  # type:Set
+        self.known_words = known_words
+
+    def run_spell_check(self):
+        """Runs spell-check on the given file.
+
+        Returns:
+            bool. True if no problematic words found, False otherwise.
+        """
+        # adding known words file if given - these words will not count as misspelled
+        if self.known_words:
+            known_words_file = open(self.known_words, 'r')
+            self.spellchecker.word_frequency.load_text_file(known_words_file)
+
+        # adding the KNOWN_WORDS to the spellchecker recognized words.
+        self.spellchecker.word_frequency.load_words(KNOWN_WORDS)
+        if self.checked_file_path.endswith('.md'):
+            self.check_md_file()
+
+        elif self.checked_file_path.endswith('.yml'):
+            with open(self.checked_file_path, 'r') as yaml_file:
+                yml_info = yaml.safe_load(yaml_file)
```

Comment:
Let's add this code to the check_yaml method

Reply:
I don't think we want to do that. The check_yml function is recursive and works by getting a dictionary.
This is simply the initialisation of the first dictionary, putting it there will add an unnecessary step in my opinion.